packages/salesforcedx-vscode-core/src/services/sdr/componentSetUtils.ts

```diff
@@ -8,20 +8,51 @@
 import { ConfigUtil } from '@salesforce/salesforcedx-utils-vscode';
 import { ComponentSet } from '@salesforce/source-deploy-retrieve';
 import { WorkspaceContext } from '../../context/workspaceContext';
+import { SfdxProjectConfig } from '../../sfdxProject';
 
-export async function setApiVersionOn(components: ComponentSet) {
+export async function setApiVersion(componentSet: ComponentSet): Promise<void> {
   // Check the SFDX configuration to see if there is an overridden api version.
   // Project level local sfdx-config takes precedence over global sfdx-config at system level.
   const userConfiguredApiVersion = await ConfigUtil.getUserConfiguredApiVersion();
-
   if (userConfiguredApiVersion) {
-    components.apiVersion = userConfiguredApiVersion;
+    componentSet.apiVersion = userConfiguredApiVersion;
     return;
   }
 
   // If no user-configured Api Version is present, then get the version from the Org.
   const orgApiVersion = await getOrgApiVersion();
-  components.apiVersion = orgApiVersion ?? components.apiVersion;
+  componentSet.apiVersion = orgApiVersion ?? componentSet.apiVersion;
+}
+
+export async function setSourceApiVersion(componentSet: ComponentSet): Promise<void> {
+  if (!componentSet) {
+    return;
+  }
+
+  // For a listing (and order of precedence) of how to retrieve the value of sourceApiVersion,
```

Comment:
I normally don't add these many comments, but since there's been so much confusion and discussion on this I think this code needs a lot of explanation on what's going on.